recipes/ucx/meta.yaml

```diff
@@ -0,0 +1,62 @@
+{% set ucx_version = "1.5.1" %}
+{% set number = "0" %}
+
+package:
+  name: ucx-split
+
+source:
+  url: https://github.com/openucx/ucx/releases/download/v{{ ucx_version }}/ucx-{{ ucx_version }}.tar.gz
+  sha256: 567119cd80ad2ae6968ecaa4bd1d2a80afadd037ccc988740f668de10d2fdb7e
+
+build:
+  skip: true  # [not linux]
+
+outputs:
+  - name: ucx-proc
+    version: 1.0.0
+    build:
+      number: 0
+      string: "{{ ucx_proc_type }}"
+    about:
+      home: https://github.com/conda-forge/ucx-feedstock
+      license: BSD-3-Clause
+      license_family: BSD
+      summary: A meta-package to select CPU or GPU UCX build.
+
+  - name: ucx
+    version: {{ ucx_version }}
+    build:
+      number: {{ number }}
+      string: "h{{ PKG_HASH }}_{{ number }}_{{ ucx_proc_type }}"
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - {{ cdt("libnl") }}
+        - {{ cdt("libibverbs-devel") }}
+        - {{ cdt("librdmacm-devel") }}
+        - {{ cdt("numactl-devel") }}
+        - automake
+        - autoconf
+        - libtool
+        - make
+        - pkg-config
+      host:
+      run_constrained:
+        - ucx-proc * {{ ucx_proc_type }}
+    script: install_ucx.sh
+    test:
+      commands:
+        - test -f "${PREFIX}/bin/ucx_info"
+        - ${PREFIX}/bin/ucx_info -v
+    about:
+      home: https://github.com/openucx/ucx
+      license: BSD-3-Clause
+      license_family: BSD
+      license_file: LICENSE
+      summary: Unified Communication X.
+
+extra:
+  recipe-maintainers:
+    - jakirkham
+    - TomAugspurger
```

Comment:
@TomAugspurger, are you ok being listed as a maintainer here?

Reply:
Should we include @akshay-venkatesh or one of the UCX maintainers here as well?  This is for the C/C++ library, correct?

Reply:
I can definitely pitch in when it comes to the process of actually building the UCX C library.

Reply:
Yeah that would be great!

Yes this is for the C/C++ library.

The Python bits get added in as a split package as part of this recipe through some other changes I have. Will hopefully be clearer once we get this through and can add PRs to feedstock.

Reply:
Have added you @Akshay-Venkatesh 🙂 If there's anyone else from UCX you think should be  added here, please let me know and will add them too.